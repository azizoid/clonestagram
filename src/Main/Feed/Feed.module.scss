.feed {
    margin-top: 2em;

    .reactionsBar {
        display: flex;
        align-content: center;
        justify-content: space-between;

        svg {
            height: 24px;
            width: 24px;
            margin: 12px;
        }
    }

    .commentsAndLikesBar {
        margin-top: 1em;
        font-size: 0.9em;

        .likes {
            font-weight: bold;
        }
    }
}
