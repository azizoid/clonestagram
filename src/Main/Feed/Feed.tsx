import React from 'react';
import { Card } from 'react-bootstrap';

import styles from './Feed.module.scss';
import { Header } from './Header/Header';

export const Feed = () => {
    return (
        <Card className={styles.feed}>
            <Header />
            <Card.Body>Body</Card.Body>
            <Card.Body>Footer</Card.Body>
        </Card>
    );
};
