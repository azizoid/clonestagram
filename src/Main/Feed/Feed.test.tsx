import React from 'react';
import { render } from '@testing-library/react';
import { Feed } from './Feed';

describe('Renders Feed', () => {
    test('renders Header', () => {
        const { container, getByText, getByRole } = render(<Feed />);

        expect(getByRole('img')).toBeInTheDocument();
        expect(getByText('azizoid')).toBeInTheDocument();
        expect(getByRole('icon')).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
