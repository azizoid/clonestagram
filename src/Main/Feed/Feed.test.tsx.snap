// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Feed renders Header 1`] = `
<div>
  <div
    class="feed card"
  >
    <div
      class="header card-header"
    >
      <div>
        <img
          class="w-32 rounded-circle img-thumbnail"
          src="https://picsum.photos/id/101/32/32"
        />
        <a
          class="profileUsernameLink username card-link"
        >
          azizoid
        </a>
      </div>
      <a
        class="threeDots card-link"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="icon"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
          />
        </svg>
      </a>
    </div>
    <div
      class="card-body"
    >
      Body
    </div>
    <div
      class="card-body"
    >
      Footer
    </div>
  </div>
</div>
`;
